dds/DCPS/DomainParticipantImpl.h

```diff
@@ -476,7 +500,7 @@ class OpenDDS_Dcps_Export DomainParticipantImpl
   /// Protect the topic collection.
   ACE_Recursive_Thread_Mutex topics_protector_;
   /// Protect the handle collection.
-  ACE_Recursive_Thread_Mutex handle_protector_;
+  mutable ACE_Thread_Mutex handle_protector_;
```

Comment:
The atomic coordination between two different data structures makes me think this ought to live in its own utility class and have a small unit test.

Reply:
Yes, but we may not be able to do that in this PR.  The pattern is essentially https://www.boost.org/doc/libs/1_66_0/libs/bimap/doc/html